cloudbuild.yaml

```diff
@@ -0,0 +1,98 @@
+substitutions:
+  _REPO: k8s-container-image-promoter
+  _GOLANGCI_LINT_VERSION: v1.13.2
+  _GOPATH: /workspace/go
+
+steps:
+- name: "gcr.io/cloud-builders/git"
+  id: setup
+  entrypoint: "bash"
+  args:
+  - "-c"
+  - |
+    set -o errexit
+    set -o nounset
+    set -o pipefail
+    set -o xtrace
+
+    shopt -s dotglob
+    # Remove the git archive that GCB auto-injects to /workspace, because it's
+    # not a real Git repository (it's just a tarball). This is required because
+    # we want to make use of git commands such as git-describe later on.
+    rm -rf /workspace/*
+
+    git \
+      clone https://github.com/GoogleCloudPlatform/${_REPO} \
+      --depth 1 \
+      --branch $BRANCH_NAME \
+      ${_GOPATH}/src/github.com/GoogleCloudPlatform/${_REPO}
+
+    # Get golangci-lint; install into /workspace/bin/golangci-lint.
+    pushd go
+    curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh \
+      | sh -s ${_GOLANGCI_LINT_VERSION}
+
+# We have to use go:debian, not go[:latest], because the latter is based on
+# alpine and does not have bash.
+- name: "gcr.io/cloud-builders/go:debian"
+  id: lint
+  waitFor:
+  - setup
+  entrypoint: "bash"
+  env:
+  - "GOPATH=${_GOPATH}"
+  dir: ${_GOPATH}/src/github.com/GoogleCloudPlatform/${_REPO}
+  args:
+  - "-c"
+  - |
+    set -o errexit
+    set -o nounset
+    set -o pipefail
+    set -o xtrace
+
+    # Lint with golangci-lint. Because megacheck/typecheck linters require the
+    # `~/go/src/gopkg.in/yaml.v2` (external dependency) to be populated in the
+    # current GOPATH, we have to bring those in.
+    go get -u github.com/golang/dep/cmd/dep
+    export PATH=${_GOPATH}/bin:$$PATH
+    dep ensure
+    ./lint.sh
+
+- name: "gcr.io/cloud-builders/bazel"
+  id: test
+  waitFor:
+  - setup
+  entrypoint: "bash"
+  dir: ${_GOPATH}/src/github.com/GoogleCloudPlatform/${_REPO}
+  args:
+  - "-c"
+  - |
+    set -o errexit
+    set -o nounset
+    set -o pipefail
+    set -o xtrace
```

Comment:
why not just call bazel directly?

Reply:
Mainly for uniformity with the other steps and also because I don't like breaking up all the arguments to bazel into a YAML list. I guess it looks a bit obtuse.

BTW I think cloudbuild.yaml might change in the future though to be yet another Prow job on some Prow cluster.

Reply:
the usual pattern in prow at least is to _not_ encode the logic in the job, fwiw

Reply:
Noted.
Mind if I make this into another issue to create some "ci" folder with all this logic in it (instead of embedding inline like this into the CI logic directly)?

Reply:
Added a new issue #5 for this.